### ISSUE TITLE
ColorSlider invisible

### ISSUE THREAD

**Issue by derjj2018** [inventwo/ioBroker.vis-inventwo#192]
If i add the color slider an choose color mode "CIE" the slider is not shown any more (invisible).

Only i set it back to RGB or HEX, the slider will be shown again..

**Comment by jkvarel** [comment]
@derjj2018 if you assign a datapoint it will show up again

**Comment by derjj2018** [comment]
If i assigne a datapoint nothing happens.
If i leave the widged (unselect it) i can it only select view menu. Then it lock like on the screenshot!
It will reappear only on change to RGB or HEX!

[cid:85d22aef-570e-46a2-ac49-410fb3bb8152]
________________________________
Von: Jannis ***@***.***>
Gesendet: Sonntag, 11. April 2021 22:04
An: inventwo/ioBroker.vis-inventwo ***@***.***>
Cc: Sebastian Jakobi ***@***.***>; Mention ***@***.***>
Betreff: Re: [inventwo/ioBroker.vis-inventwo] ColorSlider invisible (#192)


@derjj2018<https://github.com/derjj2018> if you assign a datapoint it will show up again

**Comment by jkvarel** [comment]
@derjj2018 the screenshot is not working. I can't see anything.
Can you send me the widget export so i can test it.

**Comment by derjj2018** [comment]
@jkvarel yes sure.

`[{"tpl":"i-vis-slider-color-hor","data":{"g_fixed":false,"g_visibility":false,"g_css_font_text":false,"g_css_background":false,"g_css_shadow_padding":false,"g_css_border":false,"g_gestures":false,"g_signals":false,"g_last_change":false,"visibility-cond":"==","visibility-val":1,"visibility-groups-action":"hide","iSliderRotation":"0","iColorSliderType":"CIE","iSliderHeight":"10","iSliderCorners":"100","iSliderKnobSize":"20","iSliderKnobCorners":"100","iSliderTextColor":"#ffffff","iTextSize":"12","signals-cond-0":"==","signals-val-0":true,"signals-icon-0":"/vis/signals/lowbattery.png","signals-icon-size-0":0,"signals-blink-0":false,"signals-horz-0":0,"signals-vert-0":0,"signals-hide-edit-0":false,"signals-cond-1":"==","signals-val-1":true,"signals-icon-1":"/vis/signals/lowbattery.png","signals-icon-size-1":0,"signals-blink-1":false,"signals-horz-1":0,"signals-vert-1":0,"signals-hide-edit-1":false,"signals-cond-2":"==","signals-val-2":true,"signals-icon-2":"/vis/signals/lowbattery.png","signals-icon-size-2":0,"signals-blink-2":false,"signals-horz-2":0,"signals-vert-2":0,"signals-hide-edit-2":false,"lc-type":"last-change","lc-is-interval":true,"lc-is-moment":false,"lc-format":"","lc-position-vert":"top","lc-position-horz":"right","lc-offset-vert":0,"lc-offset-horz":0,"lc-font-size":"12px","lc-font-family":"","lc-font-style":"","lc-bkg-color":"","lc-color":"","lc-border-width":"0","lc-border-style":"","lc-border-color":"","lc-border-radius":10,"lc-zindex":0,"oid":"deconz.0.Lights.20.xy"},"style":{"left":"545px","top":"440px"},"widgetSet":"vis-inventwo"}]`

![inventwo_ColorSlider_CIE](https://user-images.githubusercontent.com/44620765/114596172-9476c480-9c8f-11eb-88e3-6a29ea3e7e69.jpg)

**Comment by jkvarel** [comment]
@derjj2018 thank you.
Can you tell me what value is inside the datapoint?

**Comment by derjj2018** [comment]
@jkvarel 


RAW def of the DP:
`{
  "from": "system.adapter.deconz.0",
  "user": "system.user.admin",
  "ts": 1589645553867,
  "common": {
    "name": "undefined xy",
    "type": "array",
    "role": "color.CIE",
    "def": "0.10000, 0.10000",
    "read": true,
    "write": true
  },
  "native": {},
  "acl": {
    "object": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator",
    "state": 1636
  },
  "_id": "deconz.0.Lights.20.xy",
  "type": "state"
}`

![Screenshot 2021-04-14 100900](https://user-images.githubusercontent.com/44620765/114676330-7d74b880-9d09-11eb-88fb-0a099794d9df.png)

**Comment by jkvarel** [comment]
@derjj2018 ahh its an array? Ok, then it won't work at the moment. The datapoints from hue lamps are just normal testfields.
I need to check how i implement both options.

A workaround is to create a javascript or blockly and put the values of the array into an own datapoint

**Comment by derjj2018** [comment]
@jkvarel Thank.
But for my understanding, the Widget must not get invisible by choosing CIE (with and w/out) DP

**Comment by jkvarel** [comment]
@derjj2018 you are right. That is a bug because in that case the slider can't work because it gets an invalid value

**Comment by jkvarel** [comment]
v3.0.0: Colorslider dosn't disappear if value is invalid.
Arrays are still not working. Need to find a solution for that

**Comment by jkvarel** [comment]
Arrays should work now. I added it in any version. Please check if it is working for you
